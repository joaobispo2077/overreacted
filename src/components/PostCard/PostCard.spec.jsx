import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const createMockPostCardProps = () => {
  const postCardSut = {
    title: 'title 1',
    body: 'body 1',
    id: 1,
    cover: 'img/img.png',
  };

  return postCardSut;
};

describe('<PostCard />', () => {
  it('should render PostCard', () => {
    const mockPostCardProps = createMockPostCardProps();
    render(<PostCard {...mockPostCardProps} />);

    expect(screen.getByAltText(mockPostCardProps.title)).toHaveAttribute(
      'src',
      mockPostCardProps.cover,
    );
    expect(
      screen.getByRole('heading', {
        name: `${mockPostCardProps.id} - ${mockPostCardProps.title}`,
      }),
    ).toBeInTheDocument();
  });

  it('shoud match snapshot', () => {
    const mockPostCardProps = createMockPostCardProps();
    const { container } = render(<PostCard {...mockPostCardProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
