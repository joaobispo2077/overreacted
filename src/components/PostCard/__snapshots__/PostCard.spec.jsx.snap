// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> shoud match snapshot 1`] = `
<li
  class="post"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      1
       - 
      title 1
    </h1>
    <p>
      body 1
    </p>
  </div>
</li>
`;
